handlers/chat.js

```diff
@@ -32,6 +32,17 @@ Dota2.Dota2Client.prototype._getChannelById = function(channel_id) {
     }
 }
 
+Dota2.Dota2Client.prototype._leaveUnknownChatChannel = function(userWhoLeft) {
```

Comment:
I really like that you externalize this function, but I think making it _leaveChatChannelById would be better. The reason why is that this way we could reuse it for the leaveChatChannel function as well. This would imply having to add the debug info to the specific event handlers, which imo isn't a bad thing, since the error can probably happen in the onChatMessage and onOtherJoinedChannel handlers as well, which would require a different debug message.